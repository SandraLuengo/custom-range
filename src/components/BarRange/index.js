import BarRange from "./BarRange";

export default BarRange;