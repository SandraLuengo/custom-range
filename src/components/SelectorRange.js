import React from "react";

const SelectorRange = () => {
  return <div>selector</div>;
};

export default SelectorRange;
