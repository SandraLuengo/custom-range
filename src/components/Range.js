import React from "react";
import { SelectorRange } from "../components";

const Range = () => {
  return (
    <div>
      Range <SelectorRange />
    </div>
  );
};

export default Range;
