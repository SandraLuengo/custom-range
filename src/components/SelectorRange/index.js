import SelectorRange from "./SelectorRange";

export default SelectorRange;