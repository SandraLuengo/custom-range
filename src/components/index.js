import Range from "./Range";
import SelectorRange from "./SelectorRange";
import BarRange from "./BarRange/BarRange";

export { Range, SelectorRange, BarRange };
