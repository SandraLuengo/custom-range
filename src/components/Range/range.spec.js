import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter() });

import { Range } from "../../components";

let component = null;

describe("Range", () => {
  beforeAll(() => {
    component = mount(<Range minPrice={1} maxPrice={100} fixedType={false} />);
  });

  it("should render Exercise1", () => {
    expect(component).toMatchSnapshot();
  });

  it("it should has minPrice as prop", () => {
    expect(component.props().minPrice).toEqual(1);
  });

  it("it should has maxPrice as prop", () => {
    expect(component.props().maxPrice).toEqual(100);
  });

  it("it should has fixedType as prop", () => {
    expect(component.props().fixedType).toEqual(false);
  });

  it.skip("it should change min value", () => {

    const input = component.find("#input-left");

    input.simulate("change", { target: { value: 34 } });

    console.log(input.get(0));

    expect(component.find("#selector-left").prop("style").left).toBe("33.3333%");
  });

  it.only("it should change max value", () => {

    const input = component.find("#input-right");

    input.simulate("change", { target: { value: '50' } });

    expect(component.find("#selector-right").prop("style").left).toBe("49.4949%");
  });
});
