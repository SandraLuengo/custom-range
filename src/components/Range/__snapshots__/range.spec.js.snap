// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Range should render Exercise1 1`] = `ReactWrapper {}`;
